### PR TITLE
Feat #10584: Support Lineage for DataModel Entity

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes part : #10584

- I worked on adding Lineage in DataModel Entity

image

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx [hunk 1]
```diff
@@ -374,6 +413,86 @@ const DataModelsPage = () => {
...
+
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    fetchTabSpecificData();
+  }, [tab, dashboardDataModelFQN]);
+
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx:495]: We can reduce it to one

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx]: @chirag-madlani  updated it!

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx]: @Ashish8689 point over here was we don't need that switch case

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx]: @chirag-madlani  Actually we need it as we are fetching lineage, then activity too based on the tab. So in that case we require switch.

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx]: @Ashish8689 Lineage should be fetched within Lineage component itself

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx]: @chirag-madlani  updated as per comments!

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**